src/cmd/partition.rs

```diff
@@ -0,0 +1,223 @@
+use std::collections::{HashMap, HashSet};
+use std::collections::hash_map::Entry;
+use std::fs;
+use std::io;
+use std::path::Path;
+
+use csv;
+use regex::Regex;
+use rustc_serialize::{Decodable, Decoder};
+
+use CliResult;
+use config::{Config, Delimiter};
+use select::SelectColumns;
+use util;
+
+static USAGE: &'static str = "
+Partitions the given CSV data into chunks based on the value of a column
+
+The files are written to the output directory with filenames based on the
+values in the partition column and the `--filename` flag.
+
+Usage:
+    xsv partition [options] <column> <outdir> [<input>]
+    xsv split --help
+
+partition options:
+    --filename <filename>  A filename template to use when constructing
+                           the names of the output files.  The string '{}'
+                           will be replaced by a value based on the value
+                           of the field, but sanitized for shell safety.
+                           [default: {}.csv]
+
+Common options:
+    -h, --help             Display this message
+    -o, --output <file>    Write output to <file> instead of stdout.
+    -n, --no-headers       When set, the first row will NOT be interpreted
+                           as column names. Otherwise, the first row will
+                           appear in all chunks as the header row.
+    -d, --delimiter <arg>  The field delimiter for reading CSV data.
+                           Must be a single character. (default: ,)
+";
+
+#[derive(Clone, RustcDecodable)]
+struct Args {
+    arg_column: SelectColumns,
+    arg_input: Option<String>,
+    arg_outdir: String,
+    flag_filename: FilenameTemplate,
+    // This option is unused, both here and in the split command.
```

Comment:
Ah, probably just a mishap on my part. No need to repeat the mistakes of the past. :-)